consensus/core/src/authority_service.rs

```diff
@@ -322,7 +324,19 @@ impl<C: CoreThreadDispatcher> NetworkService for AuthorityService<C> {
         // Compute an exclusive end index and bound the maximum number of commits scanned.
         let exclusive_end =
             (end + 1).min(start + self.context.parameters.commit_sync_batch_size as CommitIndex);
-        let mut commits = self.store.scan_commits((start..exclusive_end).into())?;
+        let mut accumulated_size = 0;
+        let mut accumulated_count = 0;
+        let mut commits = self
+            .store
+            .scan_commits((start..exclusive_end).into())?
+            .into_iter()
+            .take_while(|c| {
+                accumulated_size += c.serialized().len();
+                accumulated_count += 1;
```

Comment:
It feels like this `accumulated_count` is not necessary and we should limit the size of commits that are allowed to be written in store so that we are guaranteed to atleast have one commit in store that is < `TARGET_RESPONSE_DATA_SIZE`

Reply:
Actually not sure how you can cleanly limit this, ignore the comment

Reply:
I don't think we can choose to not write some commits in store or sent them via commit sync.